<?xml version="1.0"?>
<!-- Launches the ROS-based simulation, which uses the real mpc_ros nodes talking to a PyBullet sim environment. -->

<launch>
  <arg name="config"/>

  <!-- Simulation -->
  <include file="$(find mmseq_run)/launch/simulation.launch" pass_all_args="true"/>

  <!-- Controller -->
  <node pkg="mmseq_run" type="mpc_ros.py" name="controller" output="screen" required="true" args="--config $(arg config)"/>

  <!-- TF -->
  <include file="$(find mobile_manipulation_central)/launch/thing_description.launch">
    <arg name="visual_params" value="$(find ur_description)/config/ur10/visual_parameters.yaml" />
  </include>
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="ur10_state_publisher" ns="ur10"/>
  <node name="ridgeback_state_publisher" pkg="mobile_manipulation_central" type="ridgeback_state_publisher_node.py" output="screen"/>
</launch>